compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -5361,6 +5357,29 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
         case UNWRAP_FROM_THROWABLE =>
           // js.special.unwrapFromThrowable(arg)
           js.UnwrapFromThrowable(genArgs1)
+
+        case LINKTIME_PROPERTY_BOOLEAN |
+             LINKTIME_PROPERTY_INT |
+             LINKTIME_PROPERTY_STRING =>
+          // LinkingInfo.linkTimePropertyXXX("...")
+          val arg = genArgs1
+          val tpe: jstpe.Type = code match {
+            case LINKTIME_PROPERTY_BOOLEAN => jstpe.BooleanType
+            case LINKTIME_PROPERTY_INT => jstpe.IntType
+            case LINKTIME_PROPERTY_STRING => jstpe.StringType
+          }
+          arg match {
+            case js.StringLiteral(name) =>
+              js.LinkTimeProperty(name)(tpe)
+            case _ =>
+              reporter.error(args.head.pos,
+                "The argument of linkTimePropertyXXX must be a String literal: \"...\"")
+              js.LinkTimeProperty("erroneous")(tpe)
+          }
+
+        case FILELEVEL_THIS =>
+          // js.special.fileLevelThis
+          js.JSGlobalRef(js.JSGlobalRef.FileLevelThis)
```

Comment:
For `FILELEVEL_THIS` we will want to patch its *body* rather than affect its *call sites*. That's necessary to preserve the binary API of code that actually *called* `js.special.fileLevelThis` before. See how we do it for `String.charAt` and `String.length` at
https://github.com/scala-js/scala-js/blob/867f9512b552300307c7134e29fa08340edb9a87/compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala#L2191-L2202